### ISSUE TITLE
[BUG] CUDA Out of Memory when eval model.

### ISSUE THREAD

**Issue by Crystalxd** [baichuan-inc/baichuan-7B#133]
### Required prerequisites

- [X] I have read the documentation <https://github.com/baichuan-inc/baichuan-7B/blob/HEAD/README.md>.
- [X] I have searched the [Issue Tracker](https://github.com/baichuan-inc/baichuan-7B/issues) and [Discussions](https://github.com/baichuan-inc/baichuan-7B/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [X] Consider asking first in a [Discussion](https://github.com/baichuan-inc/baichuan-7B/discussions/new).

### System information

conda environment
torch=2.0.1
transformers=4.29.2
...

### Problem description

I used A100(80G) to run the [evaluate_zh.py](https://github.com/baichuan-inc/Baichuan-7B/blob/main/evaluation/evaluate_zh.py) script for evaluating baichuan model, but it occupied abundant GPU memory up to overflow. Then I found the model loaded without eval mode, meanwhile, it inferred without no_grad.

### Reproducible example code

The Python snippets:

```python
[https://github.com/baichuan-inc/Baichuan-7B/blob/6f3ef4633a90c2d8a3e0763d0dec1b8dc11588f5/evaluation/evaluate_zh.py#L97C13-L97C13](url)
self.model = model.eval()

https://github.com/baichuan-inc/Baichuan-7B/blob/6f3ef4633a90c2d8a3e0763d0dec1b8dc11588f5/evaluation/evaluate_zh.py#L103
Add on this line:
@torch.inference_mode()
```

Command lines:

```bash

```

Extra dependencies:

```text

```

Steps to reproduce:

1.
2.
3.

### Checklist

- [X] I have provided all relevant and necessary information above.
- [X] I have chosen a suitable title for this issue.

**Comment by Guanze-Chen** [comment]
Thank you. It works!!!

**Comment by ICanFlyGFC** [comment]
Thanks!

**Comment by Guanze-Chen** [comment]
您的邮件已经收到，会尽快回复您

**Comment by Young-X** [comment]
我在训练模型过程中，脚本默认使用gpu0，怎么调换到gpu1上面？

**Comment by Guanze-Chen** [comment]
您的邮件已经收到，会尽快回复您
